### PR TITLE
Mailer

### PR DESCRIPTION
**Pull request by redwarewolf**
## Summary

- Mailer Service 

image

### CODE REVIEW

File: app/middlewares/authorizationValidator.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ exports.validate = async (req, res, next) => {
   try {
     const token = jwt.validate(req.headers.authorization);
     const user = await userService.findUserByMail(token.mail);
-    if (user.tokenEmitDate / 1000 < token.iat) {
+    if (user.tokenEmitDate / 1000 <= token.iat) {
```
Comment (picollomartin) [inline app/middlewares/authorizationValidator.js:9]: This should go in the previous PR

File: app/services/mailer.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  const transporter = nodemailer.createTransport({
+    host: 'smtp.ethereal.email',
+    port: 587,
+    secure: false,
+    auth: {
+      user: testAccount.user,
+      pass: testAccount.pass
+    }
+  });
```
Comment (picollomartin) [inline app/services/mailer.js:16]: You should have one transporter for all the service

File: app/services/mailer.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  const info = await transporter.sendMail({
+    from: '"WAlbums Admin" <admin@walbums.com>',
```
Comment (picollomartin) [inline app/services/mailer.js:19]: Make from mail configurable

File: app/services/users.js [hunk 4]
```diff
@@ -4,14 +4,20 @@ const logger = require('../logger');
...
 const createUser = data => {
   logger.info('Create User: ', data);
 
-  return User.create(data).catch(error => {
-    logger.error(error);
-    throw errors.databaseError(error.message);
-  });
+  return User.create(data)
+    .then(user => {
+      mailer.sendMail(user);
```
Comment (picollomartin) [inline app/services/users.js:14]: You're not awaiting the mail, why?

File: test/controllers/user.test.js [hunk 5]
```diff
@@ -232,7 +232,7 @@ describe('Users Controller', () => {
                 .set({ Accept: 'application/json', authorization: res.headers.authorization })
                 .then(() => {
                   expect(
-                    jwt.validate(res.headers.authorization).iat < user.tokenEmitDate / 1000
+                    jwt.validate(res.headers.authorization).iat <= Math.round(user.tokenEmitDate / 1000)
```
Comment (picollomartin) [inline test/controllers/user.test.js:235]: This also is from another PR

File: app/services/mailer.js (hunk 2 above)
Comment (picollomartin) [reply app/services/mailer.js:16]: Also, make the port and host configurable

File: app/services/users.js (hunk 4 above)
Comment (redwarewolf) [reply app/services/users.js:14]: There is an await inside the mailer function

Comment (picollomartin) [reply app/services/users.js:13]: That's not enough, you should await here too, remember than await is only a syntax sugar for promises

File: app/controllers/albums.js [hunk 6]
```diff
@@ -13,7 +14,9 @@ exports.getAlbumPhotos = (req, res, next) =>
     .catch(error => next(error));
 
 exports.buyAlbum = (req, res, next) =>
-  albumsService
-    .buyAlbum({ albumId: req.params.id, userId: req.body.user_id })
-    .then(response => res.send(response))
+  usersService
+    .getUser(req.body.user_id)
```
Comment (picollomartin) [inline app/controllers/albums.js:18]: You should have the user from the request, not the body itself !

File: app/services/mailer.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+      port: config.common.mailer.mailerPort,
+      secure: false,
+      auth: {
+        user: config.common.mailer.mailerUser,
+        pass: config.common.mailer.mailerPass
+      }
+    });
+  }
+  return transporter;
+};
```
Comment (picollomartin) [inline app/services/mailer.js:20]: You could define this as a constant directly, what advantages fives this to you?

File: app/services/users.js [hunk 8]
```diff
@@ -3,14 +3,17 @@ const errors = require('../errors');
...
 
 exports.createUser = data => {
   logger.info('Create User: ', data);
 
-  return User.create(data).catch(error => {
-    logger.error(error);
-    throw errors.databaseError(error.message);
-  });
+  return User.create(data)
+    .then(user => mailer.sendMail(user).then(() => user))
```
Comment (picollomartin) [inline app/services/users.js:12]: Mailer and user mixed again

File: test/controllers/user.test.js [hunk 9]
```diff
@@ -15,6 +16,9 @@ require('../factory/models');
 
 describe('Users Controller', () => {
   describe('/POST users', () => {
+    beforeAll(() => {
+      mailMocker.mockMailSending();
```
Comment (picollomartin) [inline test/controllers/user.test.js:20]: Check that this mock is called (or not) in some point of the tests

File: app/services/mailer.js (hunk 7 above)
Comment (redwarewolf) [reply app/services/mailer.js:20]: If I used const, It could not be mocked for some reason

**Pull request merged**